src/DeviceManager.cc

```diff
@@ -77,6 +77,7 @@ namespace ChimeraTK {
         }
       } // else do nothing. There are plenty of errors reported already: The queue is full.
       // set the error flag and notify the other threads
+      _device.setException(errMsg);
       _deviceHasError = true;
```

Comment:
Side note: I am wondering if we couldn't get rid of the `errorLock` here if we would check the `_deviceHasError` flag with an atomic `exchange()` operation. Is there a motivation for the `errorLock` other than avoiding multiple reports of the same error?